include/bout/coordinates.hxx

```diff
@@ -80,36 +87,46 @@ public:
    * Must be a better way so that Coordinates doesn't depend on Datafile
    */
   void outputVars(Datafile &file);
-  
-  Field2D dx, dy; ///< Mesh spacing in x and y
-  BoutReal dz; ///< Mesh spacing in Z
 
-  BoutReal zlength() const { return dz * nz; } ///< Length of the Z domain. Used for FFTs
+  metric_field_type dx, dy, dz; ///< Mesh spacing in x and y
+  // BoutReal dz; ///< Mesh spacing in Z
+
+  Field2D zlength() const {
+#ifdef COORDINATES_USE_3D
+    Field2D result(0., localmesh);
+    BOUT_FOR_SERIAL(i, dz.getRegion("RGN_ALL")) { result[i] += dz[i]; }
+    return result;
+#else
+    return dz * nz;
+#endif
+  } ///< Length of the Z domain. Used for FFTs
 
   /// True if corrections for non-uniform mesh spacing should be included in operators
   bool non_uniform;
-  Field2D d1_dx, d1_dy;  ///< 2nd-order correction for non-uniform meshes d/di(1/dx) and d/di(1/dy)
-  
-  Field2D J; ///< Coordinate system Jacobian, so volume of cell is J*dx*dy*dz
+  metric_field_type d1_dx, d1_dy,
+      d1_dz; ///< 2nd-order correction for non-uniform meshes d/di(1/dx) and d/di(1/dy)
```

Comment:
Just to note that this is due to clang-format not wanting to move inline comments. Unfortunately there's no setting to do actually so.

I recommend preferring writing comments entirely on their own lines because of this